modular_skyrat/modules/xenos_skyrat_redo/code/base_skyrat_xeno.dm

```diff
@@ -0,0 +1,180 @@
+/// SKYRAT MODULE SKYRAT_XENO_REDO
+
+/mob/living/carbon/alien/humanoid/skyrat
+	name = "rare bugged alien"
+	icon = 'modular_skyrat/modules/xenos_skyrat_redo/icons/big_xenos.dmi'
+	rotate_on_lying = FALSE
+	base_pixel_x = -16 //All of the xeno sprites are 64x64, and we want them to be level with the tile they are on, much like oversized quirk users
+	var/datum/action/small_sprite/skyrat_xeno/small_sprite
+	var/datum/action/cooldown/alien/skyrat/sleepytime/rest_button //There's no resting on the hud for xenos, and I don't think players want to use the ic panel
+	var/datum/action/cooldown/alien/skyrat/devolve/devolve_ability
+	mob_size = MOB_SIZE_LARGE
+	layer = LARGE_MOB_LAYER //above most mobs, but below speechbubbles
+	plane = GAME_PLANE_UPPER_FOV_HIDDEN
+	maptext_height = 64
+	maptext_width = 64
+	pressure_resistance = 200
+	var/alt_inhands_file = 'modular_skyrat/modules/xenos_skyrat_redo/icons/big_xenos.dmi'
+	var/next_evolution
+	var/datum/action/cooldown/alien/skyrat/generic_evolve/evolve_ability
+	var/unable_to_use_abilities = FALSE
+
+/mob/living/carbon/alien/humanoid/skyrat/Initialize(mapload)
+	. = ..()
+	small_sprite = new /datum/action/small_sprite/skyrat_xeno()
+	small_sprite.Grant(src)
+
+	rest_button = new /datum/action/cooldown/alien/skyrat/sleepytime()
+	rest_button.Grant(src)
+
+	devolve_ability = new /datum/action/cooldown/alien/skyrat/devolve()
+	devolve_ability.Grant(src)
+
+	if(next_evolution)
+		evolve_ability = new /datum/action/cooldown/alien/skyrat/generic_evolve()
+		evolve_ability.Grant(src)
+
+	pixel_x = -16
+
+	ADD_TRAIT(src, TRAIT_XENO_HEAL_AURA, TRAIT_XENO_INNATE)
+	real_name = "alien [caste]"
+
+/mob/living/carbon/alien/humanoid/skyrat/Destroy()
+	QDEL_NULL(small_sprite)
+	QDEL_NULL(rest_button)
+	QDEL_NULL(devolve_ability)
+	if(evolve_ability)
+		QDEL_NULL(evolve_ability)
+	return ..()
+
+/datum/action/cooldown/alien/skyrat
+	icon_icon = 'modular_skyrat/modules/xenos_skyrat_redo/icons/xeno_actions.dmi'
+	var/can_be_used_always = FALSE //Some xeno abilities block other abilities being used, sometimes we need to override that
+
+/datum/action/cooldown/alien/skyrat/IsAvailable()
+	. = ..()
+	if(!isalien(owner))
+		return FALSE
+	var/mob/living/carbon/alien/humanoid/skyrat/owner_alien = owner
+	if(!can_be_used_always)
+		if(owner_alien.unable_to_use_abilities)
+			return FALSE
+
+/datum/action/small_sprite/skyrat_xeno
+	small_icon = 'icons/obj/plushes.dmi'
+	small_icon_state = "rouny"
+
+/datum/action/cooldown/alien/skyrat/sleepytime //I don't think this has a mechanical advantage but they have cool resting sprites so...
```

Comment:
there are mechanical advantages-- they are supposed to recover faster.

Reply:
except they don't on tg code